### PR TITLE
Refactor to Android library Project

### PR DESCRIPTION
**Pull request by scottyab**
As mentioned in comments on issue #7 I've refactored the structure into an Android library project with very simple test (that needs more work) and sample project (also needs more work ideally). I've also published a temporary copy to maven central snapshots repo as an example of how this could work.  Readme also updated to account for this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,65 @@
...
+###Android Library project
+The library is in Android library project format so you can clone this project and add as a library module/project. 
+	
+###Maven Dependancy
+We've also published the library AAR file to Maven central for simple one line gradle dependancy managment. 
+
+
+```groovy
+dependencies {
+    compile 'com.scottyab:aes-crypto:0.0.1-SNAPSHOT'
```
Comment (swenson) [inline README.md]: I'm going to publish this under `com.tozny` as soon as OSS Sonatype approves the new `groupId`.

File: README.md [hunk 2]
```diff
@@ -1,16 +1,65 @@
...
 * *Key generation*: Random key generation with the updated generation code recommended for Android. If you want password-based keys, we provide functions to salt and generate them.
 * *Integrity*: Lots of people think AES has integrity checking built in. The thinking goes, "if it decrypts correctly, it was generated by the person with the private key". Actually, AES CBC allows an attacker to modify the messages. Therefore, we've also added integrity checking in the form of a SHA 256 hash.
-* *License*: The included MIT license is compatible with open source or commercial products. Tozny also offers custom support and licensing terms if your organization has different needs. Contact us at [info@tozny.com](mailto:info@tozny.com) for more details.
\ No newline at end of file
+
+
+#How to include in project?
+
+###Copy and paste
+It's a single very simple java class, [AesCbcWithIntegrity.java](https://github.com/scottyab/java-aes-crypto/blob/master/aes-crypto/src/main/java/com/tozny/crypto/android/AesCbcWithIntegrity.java) that works across most or all versions of Android. The class should be easy to paste into an existing codebase.
```
Comment (swenson) [inline README.md]: Should change link to `tozny` GitHub. :)

File: README.md [hunk 3]
```diff
@@ -1,16 +1,65 @@
...
+}
+```
+
+#Examples
+
+##Generate new key
+
+
+```java
+  AesCbcWithIntegrity.prngFixed.set(true);
```
Comment (swenson) [inline README.md]: I don't think we want to set that to true unless the PRNG has actually been fixed, right?

**Comment by swenson** [comment]
A couple of minor notes, but otherwise looks good (insofar as I know how to review all of the gradle stuff).

Going to need to be rebased against master though (as another PR was merged).

Thanks!

File: README.md (hunk 3 above)
Comment (scottyab) [inline README.md]: agreed, I misread this attribute as fix the PRNG.

**Comment by scottyab** [comment]
Made changes based on @swenson notes and after minor merge hiccup this PR should be ready for merge.

**Comment by swenson** [comment]
LGTM. I'll merge and work on shipping an artifact to Maven Central later today.

**Comment by swenson** [comment]
Okay, I tried and tried to get this to compile, but ran into some trouble.

```
Picked up _JAVA_OPTIONS: -Djava.awt.headless=true

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':sample'.
> Could not resolve all dependencies for configuration ':sample:_debugCompile'.
   > Could not find com.android.support:appcompat-v7:21.0.3.
     Searched in the following locations:
         https://repo1.maven.org/maven2/com/android/support/appcompat-v7/21.0.3/appcompat-v7-21.0.3.pom
         https://repo1.maven.org/maven2/com/android/support/appcompat-v7/21.0.3/appcompat-v7-21.0.3.jar
         https://oss.sonatype.org/content/repositories/snapshots/com/android/support/appcompat-v7/21.0.3/appcompat-v7-21.0.3.pom
         https://oss.sonatype.org/content/repositories/snapshots/com/android/support/appcompat-v7/21.0.3/appcompat-v7-21.0.3.jar
     Required by:
         com.tozny:sample:0.0.2

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 10.923 secs
```

I can't seem to get it to find the right repo to search for the `appcompat` library. Any ideas @scottyab?

Also, @SyntaxPolice -- can we enable either TravisCI or CircleCI on this repo (both free for open source), so that if we can get it to compile there, I can just publish the JARs that they build? (I don't have the group permissions to enable them.)

**Comment by scottyab** [comment]
`appcompat` is part of the Android SDK support repo (I'm not sure why these aren't somewhere like Maven central), please ensure that you've downloaded all of the support/compatibility libraries via the Android SDK Manager under the "Extras" section.

**Pull request merged**